src/utils/DMRoomMap.js

```diff
@@ -70,10 +70,47 @@ export default class DMRoomMap {
 
     _onAccountData(ev) {
         if (ev.getType() == 'm.direct') {
-            this.userToRooms = this.matrixClient.getAccountData('m.direct').getContent();
+            let userToRooms = this.matrixClient.getAccountData('m.direct').getContent();
+            const myUserId = this.matrixClient.getUserId();
+            const selfDMs = userToRooms[myUserId];
+            if (selfDMs && selfDMs.length) {
+                this._patchUpSelfDMs(userToRooms);
+                this.matrixClient.setAccountData('m.direct', userToRooms);
```

Comment:
mm, so if this does cause a loop, it could end up looping between clients as fast as the network will allow. could just do it when we start up or something so if it does loop, the clients just overwrite each other periodically rather than eating the network.

Reply:
good idea, added a flag to only do it once